js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetCheckStatusTag.tsx

```diff
@@ -1,14 +1,132 @@
-import {BaseTag, Box, Colors, Icon, Spinner, Tag} from '@dagster-io/ui-components';
+import {
+  BaseTag,
+  Box,
+  Colors,
+  Icon,
+  Spinner,
+  Tag,
+  intentToFillColor,
+} from '@dagster-io/ui-components';
+import {Link} from 'react-router-dom';
 
 import {assertUnreachable} from '../../app/Util';
+import {AssetCheckLiveFragment} from '../../asset-data/types/AssetBaseDataProvider.types';
 import {
   AssetCheckEvaluation,
   AssetCheckExecution,
   AssetCheckExecutionResolvedStatus,
   AssetCheckSeverity,
+  AssetKey,
 } from '../../graphql/types';
 import {linkToRunEvent} from '../../runs/RunUtils';
+import {TimestampDisplay} from '../../schedules/TimestampDisplay';
 import {TagActionsPopover} from '../../ui/TagActions';
+import {assetDetailsPathForAssetCheck} from '../assetDetailsPathForKey';
+
+export const CheckStatusRow = ({
+  assetCheck,
+  assetKey,
+}: {
+  assetCheck: AssetCheckLiveFragment;
+  assetKey: AssetKey;
+}) => {
+  const {executionForLatestMaterialization: execution} = assetCheck;
+
+  const CheckRow = ({
```

Comment:
You should move this component outside, otherwise it's technically a new component on every render and React will keep recreating the tree from scratch on each render.

Reply:
thanks for the call out, fixing